todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsFragment.java

```diff
@@ -59,24 +68,68 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.subscribe();
+        bind();
     }
 
     @Override
     public void onPause() {
         super.onPause();
-        mPresenter.unsubscribe();
+        unbind();
     }
 
-    @Override
-    public void setProgressIndicator(boolean active) {
+    private void bind() {
+        mSubscription = new CompositeSubscription();
```

Comment:
I would solve this slightly differently so that we wouldn't need a `Subject`. We would only subscribe to the `getStatistics()` but also add an `doOnSubscribe()` where we show the progress indicator. The `onCompleted()` or `onError()` would then remove the progress indicator. No more subject needed.